### PR TITLE
Fix rocAL Tensor build issues PR 3 - Metadata and parameter

### PR DESCRIPTION
**Pull request by fiona-gladwin**
- Fix warnings in meta_data.h file
- Fix build issues in meta nodes.
- Change parameter crop to use ROI instead of width and height vectors.
NOTE : The meta nodes will be fixed in future PR.

### CODE REVIEW

File: rocAL/rocAL/include/parameters/parameter_crop.h [hunk 1]
```diff
@@ -50,18 +50,17 @@ class CropParam
...
         y_drift_factor     = default_y_drift_factor();
     }
-    void set_image_dimensions( const std::vector<uint32_t>& in_width_, const std::vector<uint32_t>& in_height_)
+    void set_image_dimensions(RocalROI *roi)
     {
-        if(in_width_.size() != in_width.size() || in_height.size() != in_height_.size())
-            THROW("wrong input width = "+ TOSTR(in_width.size())+" or height size = "+TOSTR(in_height_.size()))
-        in_width  = in_width_;
-        in_height = in_height_;
+        if(roi == nullptr)
```
Comment (rrawther) [inline rocAL/rocAL/include/parameters/parameter_crop.h:55]: the roi is pointer to the struct passed, where as it is copied before. make sure this works in this case. Who owns the roi in this case?

File: rocAL/rocAL/source/meta_data/meta_node_crop.cpp [hunk 2]
```diff
@@ -40,8 +40,8 @@ void CropMetaNode::update_parameters(pMetaDataBatch input_meta_data, pMetaDataBa
     _crop_height = _meta_crop_param->croph_arr;
     _x1 = _meta_crop_param->x1_arr;
     _y1 = _meta_crop_param->y1_arr;
-    _input_width_val = _meta_crop_param->in_width;
-    _input_height_val = _meta_crop_param->in_height;
+    // _input_width_val = _meta_crop_param->in_width;   TODO - Commenting for now to be replaced with ROI
```
Comment (rrawther) [inline rocAL/rocAL/source/meta_data/meta_node_crop.cpp:43]: we need to move to working PR as soon as possible

File: rocAL/rocAL/source/parameters/parameter_crop.cpp [hunk 3]
```diff
@@ -57,8 +57,6 @@ void CropParam::array_init()
     croph_arr_val.resize(batch_size);
     x2_arr_val.resize(batch_size);
     y2_arr_val.resize(batch_size);
```
Comment (rrawther) [inline rocAL/rocAL/source/parameters/parameter_crop.cpp:59]: why is in_width removed? Is there anything else which holds this information

**Review by rrawther — APPROVED** [review]
Reviewing  this PR with lots of commented functionality is kind of wasting time. Can we move to full working PR soon

**Comment by shobana-mcw** [comment]
> Reviewing this PR with lots of commented functionality is kind of wasting time. Can we move to full working PR soon

If we add all changes in this PR it will be hard to review. Once the working PRs (1,2,3,4) are merged we will have a working branch. We will be issuing PR's for each feature / functionality.

**Review by LakshmiKumar23 — APPROVED** [review]

File: rocAL/rocAL/source/parameters/parameter_crop.cpp (hunk 3 above)
Comment (fiona-gladwin) [reply rocAL/rocAL/source/parameters/parameter_crop.cpp:59]: The in_width and in_height is replaced by the in_roi that is set

File: rocAL/rocAL/include/parameters/parameter_crop.h (hunk 1 above)
Comment (fiona-gladwin) [reply rocAL/rocAL/include/parameters/parameter_crop.h:55]: The ROI buffer is owned by the tensor and we are using the ptr here

**Pull request merged**